src/server.c

```diff
@@ -699,10 +739,19 @@ static void on_client_event(struct stream* stream, enum stream_event event)
 		return;
 	}
 
-	assert(client->buffer_index == 0);
+	bool cutting_text = client->cut_text.buffer != NULL;
+	void* start;
+	size_t space;
+
+	if (cutting_text == false) {
```

Comment:
nit: `cutting_text == false` is a strange way of saying `!cutting_text`. ;)

Also, I prefer boolean variable such as this one to be prefixed with `is_`. `if (!is_cutting_text)` reads much better.

Reply:
It would probably be better for the overall architecture to simply call a function here  called something like `process_cut_text_data` and return directly after it. You would then do `stream_read` inside that function. This would eliminate the need for further complicating the regular code path for `on_client_event` and it would probably also simplify `on_client_cut_text`.

Reply:
> nit: `cutting_text == false` is a strange way of saying `!cutting_text`. ;)

I was trying to be clear since you complained the other way [here](https://github.com/any1/wayvnc/pull/66#discussion_r490969328).

Reply:
This is because the result of `strcmp` is a number and not a boolean, and because of the number signifies if the compared value is equal, less than or greater, it doesn't really make sense to treat it as a boolean.

In many programming languages where types are more strongly enforced, you are not allowed to say something like `if (variable)` where `variable` is a non-boolean value.

I also think that writing null-checks as `if (!ptr)` is fine because I write a lot of them and writing `== NULL` becomes tedious after a while. ;) Also, it is kinda idiomatic for pointers.

Reply:
> This is because the result of `strcmp` is a number and not a boolean, and because of the number signifies if the compared value is equal, less than or greater, it doesn't really make sense to treat it as a boolean.

When I see (!exp), I think (exp == 0), not 'exp must be a boolean'. I've always preferred the more concise syntax.

> 
> In many programming languages where types are more strongly enforced, you are not allowed to say something like `if (variable)` where `variable` is a non-boolean value.

This is C.

> 
> I also think that writing null-checks as `if (!ptr)` is fine because I write a lot of them and writing `== NULL` becomes tedious after a while. ;) Also, it is kinda idiomatic for pointers.

Yes I do too, but then later, you say...

Reply:
>>In many programming languages where types are more strongly enforced, you are not allowed to say something like if (variable) where variable is a non-boolean value.

> This is C.

Ahh, so that's what you call all those strange incantations on the screen?


 >> I also think that writing null-checks as if (!ptr) is fine because I write a lot of them and writing == NULL becomes tedious after a while. ;) Also, it is kinda idiomatic for pointers.

>Yes I do too, but then later, you say...

I said "is fine", not "is preferred". It's fine to point out point out contradictions in order to seek a better understanding, but I feel like sometimes you're just trying to trip me up.

Anyhow, I'm not really interested in arguing about style; it's hardly productive. Most projects have a predominant or prescribed style. It is not the style itself that is important, but rather maintaining the same style throughout the code. Maybe I should write a style guide for my projects.

Reply:
> I said "is fine", not "is preferred". It's fine to point out point out contradictions in order to seek a better understanding, but I feel like sometimes you're just trying to trip me up.
> 

Sorry, I didn't mean to give that impression, my goal is to simply understand the styling here, so I can submit code that is already up to project expectations.

> Anyhow, I'm not really interested in arguing about style; it's hardly productive. Most projects have a predominant or prescribed style. It is not the style itself that is important, but rather maintaining the same style throughout the code. Maybe I should write a style guide for my projects.

Yes, style guides help but they are a lot of work too. I know wayfire recently moved to enforcing [uncrustify style template](https://github.com/WayfireWM/wayfire/blob/master/uncrustify.ini) in the CI. It is useful when you do get contributions naturally, you can simply point to documentation.